src/set_theory/surreal.lean

```diff
@@ -269,28 +439,200 @@ instance : has_lt surreal   := ⟨lt⟩
   neg               := has_neg.neg,
   add_left_neg      := by { rintros ⟨_⟩, exact quotient.sound pgame.add_left_neg_equiv },
-  add_comm          := by { rintros ⟨_⟩ ⟨_⟩, exact quotient.sound pgame.add_comm_equiv },
-  le                := (≤),
+  add_comm          := by { rintros ⟨_⟩ ⟨_⟩, exact quotient.sound pgame.add_comm_equiv }, }
+
+instance : ordered_add_comm_group surreal :=
+{ le                := (≤),
   lt                := (<),
   le_refl           := by { rintros ⟨_⟩, refl },
   le_trans          := by { rintros ⟨_⟩ ⟨_⟩ ⟨_⟩, exact pgame.le_trans },
   lt_iff_le_not_le  := by { rintros ⟨_, ox⟩ ⟨_, oy⟩, exact pgame.lt_iff_le_not_le ox oy },
   le_antisymm       := by { rintros ⟨_⟩ ⟨_⟩ h₁ h₂, exact quotient.sound ⟨h₁, h₂⟩ },
-  add_le_add_left   := by { rintros ⟨_⟩ ⟨_⟩ hx ⟨_⟩, exact pgame.add_le_add_left hx } }
+  add_le_add_left   := by { rintros ⟨_⟩ ⟨_⟩ hx ⟨_⟩, exact pgame.add_le_add_left hx },
+  .. surreal.add_comm_group }
 
 noncomputable instance : linear_ordered_add_comm_group surreal :=
-{ le_total := by rintro ⟨⟨x, ox⟩⟩ ⟨⟨y, oy⟩⟩; classical; exact
-    or_iff_not_imp_left.2 (λ h, le_of_lt oy ox (pgame.not_le.1 h)),
+{ le_total     := by rintro ⟨⟨x, ox⟩⟩ ⟨⟨y, oy⟩⟩; classical;
+                     exact or_iff_not_imp_left.2 (λ h, le_of_lt oy ox (pgame.not_le.1 h)),
   decidable_le := classical.dec_rel _,
-  ..surreal.ordered_add_comm_group }
+  .. surreal.ordered_add_comm_group }
+
+lemma nonneg_of_nsmul_pos (m : ℕ) {x : surreal} (hx : 0 < x) : 0 ≤ m • x :=
+begin
+  induction m with m hm,
+  { simp }, -- squeeze_simp fails?
+  { rw [succ_nsmul x m],
+    apply add_nonneg (le_of_lt hx) hm }
+end
+
+lemma pos_of_nsmul_pos {m : ℕ} {x : surreal} (hm : 0 < m) (hx : 0 < x) : 0 < m • x :=
+begin
+    induction m with m hm,
+    { exfalso, exact nat.lt_asymm hm hm },
+    { rw [succ_nsmul x m],
+      apply lt_add_of_pos_of_le hx (nonneg_of_nsmul_pos _ hx) }
+end
+
+lemma lt_of_nsmul_pos_lt {m : ℕ} {x y : surreal} (hm : 0 < m) (hxy : x < y) : m • x < m • y :=
+begin
+  rw ← sub_pos at *,
+  have : m • y - m • x = m • (y - x),
+    by { have := (nsmul_add y (-x) m).symm, rwa [neg_nsmul x m] at this },
+  rw this,
+  apply pos_of_nsmul_pos hm hxy,
+end
+
+lemma lt_of_gsmul_pos_lt {m : ℤ} {x y : surreal} (hm : 0 < m) (hxy : x < y) : m • x < m • y :=
+begin
+  cases m with m m,
+  { simp at *,
+    exact lt_of_nsmul_pos_lt hm hxy },
+  { exfalso,
+    rwa [← int.neg_succ_not_pos m] }
+end
+
+lemma gmul_cancel_of_pos {m : ℤ} {x y : surreal} (hm : 0 < m) (hmxy : m • x  = m • y) : x = y :=
+begin
+    contrapose hmxy,
+    cases (@ne_iff_lt_or_gt _ _ x y).1 hmxy with hmxy' hmxy',
+    { apply _root_.ne_of_lt,
+      exact lt_of_gsmul_pos_lt hm hmxy' },
+    { apply ne_of_gt,
+      change y < x at hmxy',
+      exact lt_of_gsmul_pos_lt hm hmxy' }
+end
+
+-- TODO: move this to a better location
+lemma int.pow_right_injective {x : ℤ} (h : 2 ≤ x) : function.injective (λ (n : ℕ), x ^ n) :=
```

Comment:
i know this isn't relevant to your purposes, but probably worth generalizing to 2 ≤ |x|

Reply:
The generalization of this is way more complicated, because you end up having to untangle `(-(1 + x))^m = (-(1 + x))^n`.
I can leave this as a TODO for now.

Reply:
```lean
lemma int.pow_right_injective {x : ℤ} (h : 2 ≤ x.nat_abs) : function.injective (λ (n : ℕ), x ^ n) :=
λ n m hnm, begin
  obtain pos | neg := int.nat_abs_eq x,
  { lift x to ℕ using (pos.symm ▸ int.coe_nat_nonneg _),
  norm_cast at h hnm,
  exact nat.pow_right_injective h hnm },

  dsimp only at hnm,
  rw [←neg_neg x, neg_pow, neg_pow (-x)] at hnm,
  replace neg := neg_eq_iff_neg_eq.mp neg.symm,
  generalize' hy : -x = y,
  rw [←neg_neg x, int.nat_abs_neg] at h,
  rw [hy] at *,
  lift y to ℕ using (neg.symm ▸ int.coe_nat_nonneg _),
  obtain h₁ | h₁ := neg_one_pow_eq_or ℤ n;
  obtain h₂ | h₂ := neg_one_pow_eq_or ℤ m;
  all_goals { simp only [h₁, h₂, neg_mul_eq_neg_mul_symm, one_mul, neg_inj] at hnm,
              norm_cast at h hnm,
              exact nat.pow_right_injective h hnm <|> exfalso },
  swap,
  replace hnm := hnm.symm,
  all_goals { apply int.ne_neg_of_pos _ _ hnm;
              rw int.lt_iff_le_and_ne;
              refine ⟨int.coe_nat_nonneg _, _⟩;
              norm_cast;
              change 0 ≠ _;
              symmetry;
              apply pow_ne_zero;
              linarith }
end
```

Reply:
kinda a pain in the ass but it works!

Reply:
Amazing, thanks a lot!